lib/charms/layer/apache_bigtop_base.py

```diff
@@ -130,6 +131,48 @@ def install(self):
         self.apply_patches()
         self.render_hiera_yaml()
 
+    def install_swap(self):
+        """
+        Setup swap space if needed.
+
+        Big Data apps can be memory intensive and lead to kernel OOM errors. To
+        provide a safety net against these, add swap space if none exists.
+        """
+        try:
+            swap_out = subprocess.check_output(['cat', '/proc/swaps']).decode()
+        except subprocess.CalledProcessError as e:
+            hookenv.log('Could not inspect /proc/swaps: {}'.format(e),
+                        hookenv.INFO)
+            swap_out = None
+            pass
```

Comment:
If we cannot open or read /proc/swaps (perhaps because it doesn't exist), we don't want to raise an error and fail -- we just keep going.  We'll treat `swap_out == None` as if we had no swap below.